### PR TITLE
PVC EventHandler should handle the deletion of PVC

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
From discussion on #79437, it was determined that PVC EventHandler should handle the deletion of PVC by removing finalizer.

This PR implements the handling.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @liggitt

**Comment by tedyu** [comment]
@liggitt
What do you think of the PR ?

thanks

File: pkg/controller/volume/pvcprotection/pvc_protection_controller.go [hunk 1]
```diff
@@ -256,6 +259,29 @@ func (c *Controller) pvcAddedUpdated(obj interface{}) {
 	}
 }
 
+func (c *Controller) pvcDeleted(obj interface{}) {
+	tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
```
Comment (liggitt) [inline pkg/controller/volume/pvcprotection/pvc_protection_controller.go]: this isn't correct... this should first check if obj is an instance of v1.PVC. if not, *then* check if it's a tombstone and can extract the item

File: pkg/controller/volume/pvcprotection/pvc_protection_controller.go [hunk 2]
```diff
@@ -256,6 +259,29 @@ func (c *Controller) pvcAddedUpdated(obj interface{}) {
...
+		return
+	}
+
+	if protectionutil.IsDeletionCandidate(pvc, volumeutil.PVCProtectionFinalizer) {
+		isUsed, err := c.isBeingUsed(pvc)
+		if err != nil {
+			return
+		}
+		if !isUsed {
+			c.removeFinalizer(pvc)
```
Comment (liggitt) [inline pkg/controller/volume/pvcprotection/pvc_protection_controller.go]: actually, this doesn't make sense. if the PVC was deleted, then there's no object to remove the finalizer from

**Comment by liggitt** [comment]
I was wrong in my comment... the PVC protection controller manages the finalizer on the PVC object. Once the PVC object is deleted (removed from etcd, which is what triggers the deletion callback) there is no object to remove the finalizer from.

Sorry for the confusion.

/close

**Pull request closed without merging**